### PR TITLE
Add audio feedback for rearranging list items

### PR DESCRIPTION
**Pull request by nekodex**
- [x] depends on ppy/osu-resources#200

### CODE REVIEW

File: osu.Game/Graphics/Containers/OsuRearrangeableListContainer.cs [hunk 1]
```diff
@@ -18,11 +23,44 @@ public abstract class OsuRearrangeableListContainer<TModel> : RearrangeableListC
...
+            Items.CollectionChanged += (_, args) =>
+            {
+                if (args.Action == NotifyCollectionChangedAction.Move)
+                    playSwapSample();
+            };
+        }
+
+        private void playSwapSample()
+        {
+            if (Time.Current - sampleLastPlaybackTime <= 35)
```
Comment (peppy) [inline osu.Game/Graphics/Containers/OsuRearrangeableListContainer.cs:47]: What's the case where you could make it play faster than this out of curiosity?

File: osu.Game/Graphics/Containers/OsuRearrangeableListContainer.cs [hunk 2]
```diff
@@ -18,11 +23,44 @@ public abstract class OsuRearrangeableListContainer<TModel> : RearrangeableListC
...
 
         protected abstract OsuRearrangeableListItem<TModel> CreateOsuDrawable(TModel item);
+
+        protected OsuRearrangeableListContainer()
+        {
+            Items.CollectionChanged += (_, args) =>
+            {
+                if (args.Action == NotifyCollectionChangedAction.Move)
+                    playSwapSample();
+            };
```
Comment (bdach) [inline osu.Game/Graphics/Containers/OsuRearrangeableListContainer.cs:42]: Is there a particular reason why this is done in ctor? Usual practice is to put these in `LoadComplete()`.

File: osu.Game/Graphics/Containers/OsuRearrangeableListContainer.cs (hunk 1 above)
Comment (nekodex) [reply osu.Game/Graphics/Containers/OsuRearrangeableListContainer.cs:47]: Dragging items around in a list while also scrolling the list is one particularly bad case...

with the debounce:

https://user-images.githubusercontent.com/272140/175283777-c71e5509-73c5-4af7-9ed4-ca1ca69400ea.mov

vs without:

https://user-images.githubusercontent.com/272140/175283838-2ebacd44-292e-400c-9534-18108d71f527.mov

Could probably also just do it by dragging fast enough, but the above case is pretty egregious.

**Comment by peppy** [comment]
For safety I think you want to add an early return if `!DragActive.Value`. This will ensure that external operations don't trigger the sounds.

**Review by peppy — APPROVED** [review]

**Pull request merged**